bluebottle/assignments/models.py

```diff
@@ -70,6 +88,19 @@ def required_fields(self):
 
         return fields
 
+    @property
+    def local_date(self):
+        if self.location and self.location.position:
+            tz_name = tf.timezone_at(
+                lng=self.location.position.x,
+                lat=self.location.position.y
+            )
+            tz = pytz.timezone(tz_name)
+
+            return self.date.astimezone(tz)
+        else:
+            return self.date
+
```

Comment:
So what will happen if you select 11:00 as time and then select Helsinki as location? Will then the `local_date` change to 12:00? 
What I would expect is that setting a location would change the timezone and thus changing the `date`, which still would be in UTC and then use this `local_date` to display the time in local format. 
In fact that _is_ how this method is working, but I didn't see the logic that changes the user input date to UTC based on timezone of the location.

Reply:
`self.date` is always in UTC. So this returns `self.date` in your local timezone. That is exactly what you want.

If a date is saved in the frontend, it will also be in the user's timezone (in the corresponding release branch), and will be saved to the db in UTC. So this will get you the time back as you inputted it (if the location is in your local timezone).